components/resources/library/src/desktopTest/kotlin/org/jetbrains/compose/resources/ComposeResourceTest.kt

```diff
@@ -0,0 +1,61 @@
+package org.jetbrains.compose.resources
+
+import androidx.compose.foundation.Image
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.test.junit4.createComposeRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalResourceApi::class, ExperimentalCoroutinesApi::class)
+class ComposeResourceTest {
+
+    @get:Rule
+    val rule = createComposeRule()
+
+    @Test
+    fun testMissingResource() = runTest (UnconfinedTestDispatcher()) {
+        var recompositionCount = 0
+        rule.setContent {
+            CountRecompositions(resource("missing.png").rememberImageBitmap()) {
```

Comment:
To continue our conversation about ability to load resources the blocking way. I imagine something like this:

```
BlockingLoader {

}

AsyncLoader(
  loading = { ProgressBar() }, // default { }
  error = { }, // default { }
) {
    ...
    Image(resource("img.png").rememberImageBitmap())
    ...
}

interface LoadQueue {
    fun <T> add(task: Deferred<T>)
}

val LocalLoadQueue: CompositionLocal<Loader>

// adds a load to queue. when loaded inform the parent container that we can show the content.
// Not sure, but probably we can't return ImageBitmap here. Only something that access image on a later stage.
@Composable
fun Resource.rememberImageBitmap(): Painter
```

But I don't know what to do with fonts. There is nothing like Painter there.

So, this is a big area for thoughts. Let's postpone it to the later and merge only the async version (when we resolve the minor discussions).